packages/react-dom/src/client/ReactDOMFiberOption.js

```diff
@@ -17,14 +18,23 @@ function flattenChildren(children) {
 
   // Flatten children and warn if they aren't strings or numbers;
   // invalid types are ignored.
-  // We can silently skip them because invalid DOM nesting warning
-  // catches these cases in Fiber.
   React.Children.forEach(children, function(child) {
     if (child == null) {
       return;
     }
     if (typeof child === 'string' || typeof child === 'number') {
       content += child;
+      return;
+    }
+    if (__DEV__) {
+      // We do not have HostContext here
+      // but we can put some parent information at least
+      // Also this cause a bit different message than it was previously
```

Comment:
Nit: “a bit different message” would become meaningless after merging this PR. Let’s ensure any comments are helpful to future readers.